### PR TITLE
(#150) Back to Top tests 

### LINKED ISSUE #150: Story: Back to top tests
## As a user, I want to be able to use back to top arrow for easy page navigation

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario Outline: Back to top arrow on desktop
  Given user is navigating to "<url>"
  When user scrolls 600 px down
  Then back-to-top arrow appears with text "TOP" and href "#top"
 When user scrolls up 100 px
Then back-to-top arrow is not displayed

Examples:
|url|
## TODO add 1 url for each content 
```

```gherkin
Scenario Outline: Back to top arrow  does not appear on mobile
Given screen breakpoint is set to "mobile"
  When user is navigating to "<url>"
  When user scrolls 600 px down
Then back-to-top arrow is not displayed

Examples:
|url|
## TODO add 1 url for each content 
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*
The system shall not display the "Back To Top" option when the user is viewing the site in mobile size
The system shall allow the Back to Top option to appear when user has scrolled to 600 px down the screen
The system shall not display the Back to Top option when user is at the top of the screen on desktop
* > Should live in crosscutting folder
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Added feature file
Added tests
Closes #150

### CODE REVIEW

**Comment by alinai** [comment]
Test Results:
image

@KateMashkinaNIH , please review.

File: cypress/integration/Crosscutting/BackToTopTest.feature [hunk 1]
```diff
@@ -0,0 +1,91 @@
+Feature: As a user, I want to be able to use back to top arrow for easy page navigation
+
+    Scenario Outline: Back to top arrow on desktop and tablet on the English site
+        Given user is navigating to "<url>"
+        Given screen breakpoint is set to "<screenBreakpoint>"
+        When user scrolls 600 px down
```
Comment (kate-mashkina) [inline cypress/integration/Crosscutting/BackToTopTest.feature:6]: this should be 601 px - at 600 it is still not visible

Comment (kate-mashkina) [reply cypress/integration/Crosscutting/BackToTopTest.feature:6]: pls update this value everywhere

File: cypress/integration/Crosscutting/BackToTopTest/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/// <reference types="Cypress" />
+import { When, Then, And } from "cypress-cucumber-preprocessor/steps";
+
+When('user scrolls 600 px down', () => {
+    cy.scrollTo(0, 600);
```
Comment (kate-mashkina) [inline cypress/integration/Crosscutting/BackToTopTest/index.js:5]: the number should not be hardcoded - please pass it as parameter

File: cypress/integration/Crosscutting/BackToTopTest/index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+});
+
+Then('back-to-top arrow appears with text {string} and href {string}', (textTop, hrefTop) => {
+    cy.get('div.site-footer__container > a').should('have.text', textTop).and('have.attr', 'href', hrefTop);
+});
+
+When('user scrolls up 100 px', () => {
+    cy.scrollTo(10, 0);
```
Comment (kate-mashkina) [inline cypress/integration/Crosscutting/BackToTopTest/index.js:13]: the number should not be hardcoded - please pass it as parameter

File: cypress/integration/Crosscutting/BackToTopTest/index.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+/// <reference types="Cypress" />
+import { When, Then, And } from "cypress-cucumber-preprocessor/steps";
+
+When('user scrolls 600 px down', () => {
+    cy.scrollTo(0, 600);
+});
+
+Then('back-to-top arrow appears with text {string} and href {string}', (textTop, hrefTop) => {
+    cy.get('div.site-footer__container > a').should('have.text', textTop).and('have.attr', 'href', hrefTop);
```
Comment (kate-mashkina) [inline cypress/integration/Crosscutting/BackToTopTest/index.js:9]: this is not checking for visibility - that is why your test for 600 px was passing where in fact it should not have

make it 
  cy.get('.back-to-top').should('be.visible').and('have.text', textTop).and('have.attr', 'href', hrefTop);
});

File: cypress/integration/Crosscutting/BackToTopTest/index.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+});
+
+When('user scrolls up 100 px', () => {
+    cy.scrollTo(10, 0);
+});
+
+Then('back-to-top arrow is not displayed', () => {
+    cy.get('div.site-footer__container > a').should('not.be.visible');
```
Comment (kate-mashkina) [inline cypress/integration/Crosscutting/BackToTopTest/index.js:17]: make a locator .back-to-top

**Review by kate-mashkina — CHANGES_REQUESTED** [review]

**Comment by alinai** [comment]
@KateMashkinaNIH , updated per your review. Test results after updating with reviews:

image

File: cypress/integration/Crosscutting/BackToTopTest.feature (hunk 1 above)
Comment (alinai) [reply cypress/integration/Crosscutting/BackToTopTest.feature]: done

File: cypress/integration/Crosscutting/BackToTopTest/index.js (hunk 2 above)
Comment (alinai) [reply cypress/integration/Crosscutting/BackToTopTest/index.js]: done

File: cypress/integration/Crosscutting/BackToTopTest/index.js (hunk 3 above)
Comment (alinai) [reply cypress/integration/Crosscutting/BackToTopTest/index.js]: done

File: cypress/integration/Crosscutting/BackToTopTest/index.js (hunk 4 above)
Comment (alinai) [reply cypress/integration/Crosscutting/BackToTopTest/index.js]: done

File: cypress/integration/Crosscutting/BackToTopTest/index.js (hunk 5 above)
Comment (alinai) [reply cypress/integration/Crosscutting/BackToTopTest/index.js]: done

**Review by kate-mashkina — APPROVED** [review]

**Pull request merged**